src/dbnode/encoding/arrow/tile/counter_resets_downsampler_test.go

```diff
@@ -0,0 +1,383 @@
+			given: []float64{3, 5},
+			want:  []DownsampledValue{{0, 3}, {1, 5}},
+		},
+		{
+			name:  "two identical values",
+			given: []float64{5, 5},
+			want:  []DownsampledValue{{1, 5}},
+		},
+		{
+			name:  "two values with reset",
+			given: []float64{3, 1},
+			want:  []DownsampledValue{{0, 3}, {1, 1}},
+		},
+		{
+			name:  "second value equal to first, then reset",
+			given: []float64{2, 2, 1},
+			want:  []DownsampledValue{{0, 2}, {2, 1}},
+		},
+		{
+			name:  "three values, no reset",
+			given: []float64{3, 4, 5},
+			want:  []DownsampledValue{{0, 3}, {2, 5}},
+		},
+		{
+			name:  "three identical values",
+			given: []float64{5, 5, 5},
+			want:  []DownsampledValue{{2, 5}},
+		},
+		{
+			name:  "three values, reset after first",
+			given: []float64{3, 1, 5},
+			want:  []DownsampledValue{{0, 3}, {1, 1}, {2, 5}},
+		},
+		{
+			name:  "three values, reset after second",
+			given: []float64{3, 5, 1},
+			want:  []DownsampledValue{{0, 3}, {1, 5}, {2, 1}},
+		},
+		{
+			name:  "three values, two resets",
+			given: []float64{5, 3, 2},
+			want:  []DownsampledValue{{0, 5}, {1, 8}, {2, 2}},
+		},
+		{
+			name:  "four values, reset after first",
+			given: []float64{3, 1, 4, 5},
+			want:  []DownsampledValue{{0, 3}, {1, 1}, {3, 5}},
+		},
+		{
+			name:  "four values, reset after second (A)",
+			given: []float64{3, 4, 1, 5},
+			want:  []DownsampledValue{{0, 3}, {1, 4}, {2, 1}, {3, 5}},
+		},
+		{
+			name:  "four values, reset after second (B)",
+			given: []float64{3, 4, 1, 4},
+			want:  []DownsampledValue{{0, 3}, {1, 4}, {2, 1}, {3, 4}},
+		},
+		{
+			name:  "four values, reset after second (C)",
+			given: []float64{3, 4, 1, 2},
+			want:  []DownsampledValue{{0, 3}, {1, 4}, {3, 2}},
+		},
+		{
+			name:  "four values, reset after third",
+			given: []float64{3, 4, 5, 1},
+			want:  []DownsampledValue{{0, 3}, {2, 5}, {3, 1}},
+		},
+		{
+			name:  "four values, two resets (A)",
+			given: []float64{3, 1, 5, 4},
+			want:  []DownsampledValue{{0, 3}, {2, 8}, {3, 4}},
+		},
+		{
+			name:  "four values, two resets (B)",
+			given: []float64{5, 2, 1, 4},
+			want:  []DownsampledValue{{0, 5}, {1, 7}, {3, 4}},
+		},
+		{
+			name:  "four values, two resets (C)",
+			given: []float64{5, 2, 2, 1},
+			want:  []DownsampledValue{{0, 5}, {2, 7}, {3, 1}},
+		},
+		{
+			name:  "four values, two resets (D)",
+			given: []float64{3, 5, 2, 1},
+			want:  []DownsampledValue{{0, 3}, {2, 7}, {3, 1}},
+		},
+		{
+			name:  "reset between two equal values",
+			given: []float64{4, 3, 4},
+			want:  []DownsampledValue{{0, 4}, {1, 3}, {2, 4}},
+		},
+		{
+			name:  "four values, three resets",
+			given: []float64{9, 7, 4, 1},
+			want:  []DownsampledValue{{0, 9}, {2, 20}, {3, 1}},
+		},
+		{
+			name:  "five values, two resets (A)",
+			given: []float64{3, 1, 2, 5, 4},
+			want:  []DownsampledValue{{0, 3}, {3, 8}, {4, 4}},
+		},
+		{
+			name:  "five equal values",
+			given: []float64{1, 1, 1, 1, 1},
+			want:  []DownsampledValue{{4, 1}},
+		},
```

Comment:
Can you add cases for `[]float64{1,2,3,4,5}` and `[]float64{5,4,3,2,1}`?

Reply:
Sure, will add.